domain/src/main/kotlin/hr/asee/android/template/domain/repository/FilterRepository.kt

```diff
@@ -0,0 +1,8 @@
+package hr.asee.android.template.domain.repository
+
+import java.time.LocalDate
+
+interface FilterRepository {
```

Comment:
Ovo će se vjerojatno raditi posebno za svaki objekt pa će se filtracija radit odvojeno za rezervacije, seekinge, offere itd, a to će biti u posebnim repozitorijima, pa ovo ne treba. Filter (datumi u ovom slucaju) općenito nije neka domena podataka, pa ne treba imati svoj repozitorij
Najdalje što bih išao je use case, npr. `FilterByDateUseCase` pa bi se tamo pozivale filtracije svih "domena" npr. rezervacije, seekinge itd., pa bi tamo izvršio ove provjere validnosti što imaš u `FilterRepositoryImpl`